develop/core/middleware.py

```diff
@@ -0,0 +1,35 @@
+from django.contrib.sites.models import Site
+from django.db.models import Q
+from django.shortcuts import redirect
+from django.urls import resolve
+from django.utils import timezone
+from user_notifications.models import DisplayType, Notification
+
+class NotificationRedirctMiddleware:
+    def __init__(self, get_response):
+        self.get_response = get_response
+        # One-time configuration and initialization.
+
+    def __call__(self, request):
+        # Code to be executed for each request before
+        # the view (and later middleware) are called.
+        site = Site.objects.get_current()
```

Comment:
i dont think get_current() will return correct results on TC- should probably check if there is a site attribute on the request object and grab request.site if so, defaulting to site.objects.get_current() if not

Reply:
```
    if getattr(request, "site", None):
        site = request.site
    else:
        site = Site.objects.get_current()
```